engine/src/main/java/org/terasology/rendering/assets/texture/AWTTextureFormat.java

```diff
@@ -120,4 +69,64 @@ public TextureData load(ResourceUrn urn, List<AssetDataFile> inputs) throws IOEx
     public PathMatcher getFileMatcher() {
         return path -> super.getFileMatcher().matches(path) && pathMatcher.matches(path);
     }
+
+    /**
+     * Converts a BufferedImage into a TextureData
+     *
+     * @param image any type of BufferedImage
+     * @param filterMode any method used to determine the texture color for a texture mapped pixel
+     * @return a TextureData that contains the data from buffered image
+     */
+    public static TextureData convertToTextureData(final BufferedImage image, Texture.FilterMode filterMode) throws IOException {
+        final WritableRaster raster = image.getRaster();
+        final DataBufferByte dbf = (DataBufferByte) raster.getDataBuffer();
+        byte[] data = dbf.getData();
+        final int stride = data.length / (image.getWidth() * image.getHeight());
+        // Sanity check
+        if (stride > 4) {
+            throw new IOException("Image data makes no sense (got "
```

Comment:
This will output error messages like "got 779012535, expected 779012530" which are a bit confusing. I'd mention

- what kind of units are we expecting to get (presumably # of pixels)

- that the fact this failed is not just because the data made no sense, but because the data was larger than we anticipated

Reply:
Hm, okay what about? "Image data are corrupted! We expect less number of pixels: #123 instead of #125."

Reply:
"Image data *is* corrupted! ..."